packages/react-router/lib/components.tsx

```
@@ -584,7 +584,7 @@ export function createRoutesFromChildren(
       handle: element.props.handle,
     };
 
-    if (element.props.children) {
+    if (element.props.children && !route.index) {
```

I think it would be better to use an `invariant` here so that the error is identified and surfaced to the user, instead of silently skipping children in a mismatched scenario?

---

Add the invariant.